src/main/java/io/reactivesocket/cli/ObservableIO.java

```diff
@@ -26,40 +26,44 @@
 
 import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.util.concurrent.ThreadFactory;
 
 public class ObservableIO {
-  /**
-   * Return a publisher for consuming each line from System.in.
-   */
-  public static Publisher<Payload> systemInLines() {
-    SyncOnSubscribe<BufferedReader, Payload> s = new SyncOnSubscribe<BufferedReader, Payload>() {
-      @Override protected BufferedReader generateState() {
-        return new BufferedReader(new InputStreamReader(System.in));
-      }
+    /**
+     * Return a publisher for consuming each line from System.in.
+     * @param inputStream
+     */
+    public static Publisher<Payload> lines(InputStream inputStream) {
+        SyncOnSubscribe<BufferedReader, Payload> s = new SyncOnSubscribe<BufferedReader, Payload>() {
+            @Override
+            protected BufferedReader generateState() {
+                return new BufferedReader(new InputStreamReader(inputStream));
+            }
 
-      @Override protected BufferedReader next(BufferedReader state, Observer<? super Payload> observer) {
-        try {
-          String line = state.readLine();
+            @Override
+            protected BufferedReader next(BufferedReader state, Observer<? super Payload> observer) {
+                try {
+                    String line = state.readLine();
 
-          if (line == null) {
-            observer.onCompleted();
-          } else {
-            observer.onNext(new PayloadImpl(line, ""));
-          }
-        } catch (IOException e) {
-          observer.onError(e);
-        }
+                    if (line == null) {
```

Comment:
No I think its just pointing out that after bulk reformatting with the Default code style in Intellij, those are space characters it inserted.